test/utils.jl

```diff
@@ -70,31 +74,38 @@ function isolate(fn::Function; loaded_depot=false)
         Base.ACTIVE_PROJECT[] = old_active_project
         cd(old_working_directory)
         Pkg.REPLMode.TEST_MODE[] = false # reset unconditionally
+        Pkg.Registry.DEFAULT_REGISTRIES[1].path = old_general_registry_path
         Pkg.Registry.DEFAULT_REGISTRIES[1].url = old_general_registry_url
+        Pkg.Registry.DEFAULT_REGISTRIES[1].linked = old_general_registry_linked
     end
 end
 
-function temp_pkg_dir(fn::Function;rm=true)
+function temp_pkg_dir(fn::Function;rm=true, linked_reg=true)
     old_load_path = copy(LOAD_PATH)
     old_depot_path = copy(DEPOT_PATH)
     old_home_project = Base.HOME_PROJECT[]
     old_active_project = Base.ACTIVE_PROJECT[]
     old_general_registry_url = Pkg.Registry.DEFAULT_REGISTRIES[1].url
+    old_general_registry_path = Pkg.Registry.DEFAULT_REGISTRIES[1].path
+    old_general_registry_linked = Pkg.Registry.DEFAULT_REGISTRIES[1].linked
     try
         # Clone the registry only once
-        generaldir = joinpath(@__DIR__, "registries", "General")
-        if !isdir(generaldir)
-            mkpath(generaldir)
-            LibGit2.with(Pkg.GitTools.clone(Pkg.Types.Context().io,
-                                            "https://github.com/JuliaRegistries/General.git",
-                generaldir)) do repo
+        if !isdir(REGISTRY_DIR)
+            mkpath(REGISTRY_DIR)
+            Base.shred!(LibGit2.CachedCredentials()) do creds
+                LibGit2.with(Pkg.GitTools.clone(Pkg.Types.Context().io,
+                                                "https://github.com/JuliaRegistries/General.git",
+                    REGISTRY_DIR, credentials = creds)) do repo
+                end
```

Comment:
Cloning the registry to two different cache locations for `isolate` and `temp_pkg_dir` seemed unnecessary?